app/src/main/java/com/lehtimaeki/askold/landingscreen/LandingScreenFragment.kt

```diff
@@ -1,176 +1,198 @@
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
-import androidx.recyclerview.widget.GridLayoutManager
-import androidx.recyclerview.widget.RecyclerView
 import com.lehtimaeki.askold.ColorPalettes
 import com.lehtimaeki.askold.FullscreenActivity
 import com.lehtimaeki.askold.FullscreenActivity.Companion.ICON_SET_EXTRA_ID
-import com.lehtimaeki.askold.FullscreenActivity.Companion.IS_LARGE_CARD_MODE_ID
 import com.lehtimaeki.askold.R
-import com.lehtimaeki.askold.databinding.LandingScreenFragmentBinding
-import com.lehtimaeki.askold.delegates.viewBinding
 import com.lehtimaeki.askold.iconset.IconSet
 
-class LandingScreenFragment : Fragment(R.layout.landing_screen_fragment) {
-
-    private val binding by viewBinding(LandingScreenFragmentBinding::bind)
+class LandingScreenFragment : Fragment() {
 
     companion object {
         fun newInstance() = LandingScreenFragment()
     }
 
     private val viewModel: LandingScreenViewModel by viewModels()
-    private val adapter = IconSetAdapter()
-
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-
 
-        val metrics = resources.displayMetrics
-        val width = metrics.widthPixels
-
-        val numberOfColumns = (width / resources.getDimension(R.dimen.minimal_tile_size)).toInt()
-
-        val layoutManager = GridLayoutManager(context, numberOfColumns)
-        layoutManager.spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
-            override fun getSpanSize(position: Int): Int {
-                return if (adapter.dataSet.get(position).iconSet != null) {
-                    1
-                } else {
-                    numberOfColumns
-                }
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setContent {
+                val iconSets = viewModel.iconSets.observeAsState().value
```

Comment:
Use the `getValue` extension to use property delegation

```suggestion
                val iconSets by viewModel.iconSets.observeAsState()
```